### PR TITLE
Security suite

### PR DESCRIPTION
**Pull request by leoleljak**
# Description

Generalised wrapper for IOSSecuritySuite which is used on some projects. The wrapper is mostly responsible for masking the lib behind not commonly used security keywords (e.g. pin, jailbreak, debugger, etc.) that attackers could easily override. Other than that, the wrapper will periodically check (every 3 sec on the jb device) if the debugger is attached and it will handle the logic for setting the blur overlay when the app enters the background. 

Some code (regarding Locker) is commented out so that the developer can either uncomment it or remove depending on the need of the project.

# Checklist:

- [x] Have checked there is no same component already in catalog.
- [x] Have created a sufficient example or wrote tests for it.
- [x] Code is structured, well written using standard Swift coding style.
- [x] All public methods and properties have meaningful and concise documentation.
- [x] Used `public` modifier for all method and properties that could be changed from user of your feature, and `private` for internal properties.
- [x] Removed any reference to the project that piece of code was created in.
- [x] Reduced the dependencies to minimum.

### CODE REVIEW

File: Sources/Utilities/Security/SecurityProtectionManager/SecurityProtectionManager.swift
Comment (jabou) [inline Sources/Utilities/Security/SecurityProtectionManager/SecurityProtectionManager.swift:107]: Is this something that we could configure? Something like `protectAppInBackground(window: UIWindow?, type: BlurProtectionLevel)` where `BlurProtectionLevel` can have something like `alwaysBlur`, `blurOnBackground`, `securityStateBase`... something like this. Then the user can decide if blur should be done on any inactivity, only when the app goes to the background or depending on the system state (as it is implemented now).

File: Sources/Utilities/Security/SecurityProtectionManager/SecurityProtectionManager.swift
Comment (jabou) [inline Sources/Utilities/Security/SecurityProtectionManager/SecurityProtectionManager.swift:150]: Let's add TODO here. That way, we can raise awareness of this method if someone just copy/paste those files.
```suggestion
    func checkSystemChange() {
        guard SecurityService.ServiceStatus.firstLevel == .offline else { return }
        // Device is Jailbroken
        // TODO: Add custom logic for showing alert or restricting access to the application.
    }
```

File: Sources/Utilities/Security/SecurityProtectionManager/SecurityProtectionManager.swift
Comment (jabou) [inline Sources/Utilities/Security/SecurityProtectionManager/SecurityProtectionManager.swift:197]: Let's remove biometrics extension :)

File: Sources/Utilities/Security/SecurityProtectionManager/SecurityProtectionManager.swift
Comment (jabou) [inline Sources/Utilities/Security/SecurityProtectionManager/SecurityProtectionManager.swift:121]:
```suggestion
    // TODO: Add biometric check if app is using biometric. Otherwise, remove this method.
    /// Checks biometrics state every time app becomes active.
    ///
    func biometricChangesCheck() {
        UIApplication.rx.didBecomeActive
            .subscribe(onNext: { [unowned self] in
                // Logic implementation
            })
            .disposed(by: disposeBag)
    }
```

**Review by jabou — APPROVED** [review]
Looks ok to me. Just a few things that should be removed/updated.

Before we merge this:
- please resolve all comments
- Update readme (add info regarding this manager under the "Features" -> Utilities)
- get at least one extra PR review from someone other than me :)

File: Catalog/Application/AppDelegate.swift
Comment (nikolamajcen) [inline Catalog/Application/AppDelegate.swift:20]: I would swap lines `20` and `21` to have `securityProtectionManager` logic one after another, but still, it can stay like that.

**Review by nikolamajcen — APPROVED** [review]
All good 👍

**Pull request merged**
